lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -28,6 +28,22 @@ async function runInSeriesOrParallel(values, promiseMapper, runInSeries) {
   }
 }
 
+/**
+ * Returns true if the script was created via our own calls
+ * to Runtime.evaluate.
+ * @param {LH.Crdp.Debugger.ScriptParsedEvent} script
+ */
+function isLighthouseRuntimeEvaluateScript(script) {
+  // Scripts created by Runtime.evaluate that run on the main session/frame
+  // result in an empty string for the embedderName.
+  if (!script.embedderName) return true;
```

Comment:
this first check is not required for correctness, but serves as documenting the behavior. it also could catch (most) of errant usages of Runtime.evaluate directly (which won't have our sourceURL comment necessarily), which could occur from an enterprising LH user.

Reply:
this isn't guaranteed. i just tried out this page and the inner function has no embedderName

```html
<!doctype html>
<h1>hi</h1>
<script>
new Function(`

document.body.style.backgroundColor = '#'+Math.floor(Math.random()*16777215).toString(16);

`)();
</script>
```

yeah actually theverge is FULL of `embedderName  === ''` scripts that arent' from LH. evals im guessing?

dunno how we got started this this assumption, but it doesnt hold up

Reply:
I had a hunch that was the case. See comments in JsUsage. It mentions the anonymous/eval bit making the url property blank. That's the same as the embedder name so it makes sense.

Let's consider loosening this to include eval scripts in a follow-up (separate from this change and from loosening to include OOPIFs)

Reply:
#13770